clustermesh/docs: misc improvements and extend the intro doc (#46021)
cilium-team/cilium: Documentation/network/clustermesh/intro.rst

* :ref:`Service affinity <gs_clustermesh_service_affinity>` controls to
  prefer local or remote backends

See :ref:`gs_clustermesh` for instructions on how to set up Cluster Mesh.

Architecture
============

Cluster Mesh is built around a per-cluster control plane hosted by
``clustermesh-apiserver`` pods. These pods expose the local cluster state to
remote clusters and synchronize remote cluster state into the local cluster.

Cilium agents use local Kubernetes API server state and remote Cluster Mesh
state in a similar way to program the local datapath. Pod-to-pod traffic is
forwarded directly between nodes, including across cluster boundaries, without
requiring any additional proxy or gateway.

The following diagram shows the main Cluster Mesh components and how two
clusters exchange state:

.. image:: images/clustermesh-architecture.svg
   :align: center

A ``clustermesh-apiserver`` pod typically contains the following containers:

* ``etcd``: Stores the local cluster state that is exposed to remote clusters
  over mTLS, typically through a LoadBalancer or NodePort service. This embedded
  etcd is simple to operate because it is a single instance and does not require

>   persistent storage as it is rebuilt from kube-apiserver on startup.

Reviewer: Nit. s/it is/the entire state is/

Multi-Cluster (Cluster Mesh)
############################

Multi-cluster Kubernetes setups are often adopted for reasons like fault
isolation, scalability, and geographical distribution. This approach can lead

> to networking complexity. Traditional networking models, in this context,

Reviewer 2: ```suggestion
to networking complexity. With such multi-cluster setups, traditional networking models
```

Not sure that inserting “in this context” within commas in the middle is idiomatic in English (but I could be wrong); and let's maybe make “this context” more explicit.

Author: That looks more natural to me thanks!

FYI these sentence are from https://cilium.io/use-cases/cluster-mesh/ (will probably send a followup there with the same edit)

* Pod-to-pod connectivity between clusters within a flat IP address space
* Cluster-aware :ref:`network policy enforcement <gs_clustermesh_network_policy>`
* Cross-cluster service discovery and load-balancing with
  :ref:`Global Services <gs_clustermesh_services>` or
  :ref:`MCS-API <gs_clustermesh_mcsapi>`
* :ref:`Service affinity <gs_clustermesh_service_affinity>` controls to
  prefer local or remote backends

See :ref:`gs_clustermesh` for instructions on how to set up Cluster Mesh.

Architecture
============

Cluster Mesh is built around a per-cluster control plane hosted by
``clustermesh-apiserver`` pods. These pods expose the local cluster state to
remote clusters and synchronize remote cluster state into the local cluster.

Cilium agents use local Kubernetes API server state and remote Cluster Mesh
state in a similar way to program the local datapath. Pod-to-pod traffic is
forwarded directly between nodes, including across cluster boundaries, without
requiring any additional proxy or gateway.

The following diagram shows the main Cluster Mesh components and how two
clusters exchange state:

.. image:: images/clustermesh-architecture.svg

>    :align: center

Reviewer 2: I'd like if we could comment somewhere that the source is available in the repo for the picture. I don't think we can add a comment in the source itself, it looks like JSON to me, and we can't add a comment on the rendered SVG obviously. So maybe add a comment here in the RST where you include the picture, to mention that the source is available alongside the SVG? This way, next time someone needs to edit it, they get a better chance to find it?

Author: Sure, thanks for the suggestion! I added a comment in the rst as you suggested.
